lib/sparql.jison

```diff
@@ -754,6 +827,28 @@ GraphRefAll
 QuadPattern
     : '{' TriplesTemplate? QuadsNotTriples* '}' -> $2 ? unionAll($3, [$2]) : unionAll($3)
     ;
+QuadPatternGround
+    : QuadPattern
+    {
+      ensureNoVariables($1);
+      $$ = $1
+    }
+    ;
+QuadPatternNoBnodes
+    : QuadPattern
+    {
+      ensureNoBnodes($1);
+      $$ = $1
+    }
+    ;
+QuadPatternGroundNoBnodes
+    : QuadPattern
+    {
+      ensureNoVariables($1);
+      ensureNoBnodes($1);
+      $$ = $1
+    }
+    ;
```

Comment:
Can we perhaps move those checks inline, to avoid creating new rules?
Every rule comes with quite a bit of generated code (unfortunately).